widget/embedded/src/store/slices/data.ts

```diff
@@ -346,4 +324,50 @@ export const createDataSlice: StateCreator<
       throw error;
     }
   },
+  setSupportedTokens: (options) => {
+    const tokens = get()._tokensMapByBlockchainName;
+    let nextState: Record<TokenHash, Token> = {};
+    if (Array.isArray(options.data.tokens)) {
+      options.data.tokens.forEach((token) => {
+        const tokenHash = createTokenHash(token);
+        const tokenFromMeta = tokens?.[token.blockchain]?.[tokenHash];
+        if (tokenFromMeta) {
+          nextState[tokenHash] = tokenFromMeta;
+        }
+      });
+    } else {
+      if (!options.data.blockchains) {
```

Comment:
Maybe it's better to use `else if` here.

Reply:
it isn't working here

Reply:
foreEach on `options.data.blockchains` worked only if `options.data.blockchains` existed.